src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1447,7 +1446,7 @@ static bool processSubscriptions
     }
 
     /* Check 2: String Filter */
-    if (!trigs->stringFilter.match(notifyCerP))
+    if (trigs->stringFilterP && !trigs->stringFilterP->match(notifyCerP))
```

Comment:
In addition, what about 

trigs->stringFilterP != NULL

I know... NULL is 0 and it works as it is :) However, doing that will make this code more readable.